### PR TITLE
BE: Make it possible to hide stacktraces in HTTP responses #536

### PR DESCRIPTION
**Pull request by mrlittle113**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)
Create a flag in application.yml to enable or disable stacktrace include in error responses

**Is there anything you'd like reviewers to focus on?**
Where to put the config on, this is my first PR so I am not familiar with the project structure!

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/kafbat/kafka-ui/blob/main/.github/CONTRIBUTING.md) and [Code of Conduct](https://github.com/kafbat/kafka-ui/blob/main/.github/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: api/src/main/resources/application.yml [hunk 1]
```diff
@@ -19,3 +19,7 @@ logging:
     reactor.netty.http.server.AccessLog: INFO
     org.hibernate.validator: WARN
 
+web:
```
[the remark this answers is not in the document]
Comment (mrlittle113) [reply api/src/main/resources/application.yml:22]: Agree, look much better!

File: api/src/main/java/io/kafbat/ui/exception/GlobalErrorWebExceptionHandler.java [hunk 2]
```diff
@@ -132,13 +136,21 @@ private Mono<ServerResponse> render(ResponseStatusException exception, ServerReq
...
+        .stackTrace(getStackTrace(exception));
     return ServerResponse
         .status(exception.getStatusCode())
         .contentType(MediaType.APPLICATION_JSON)
         .bodyValue(response);
   }
 
+  private String getStackTrace(Throwable exception) {
+    if (!includeStacktraceInException) {
+      return "";
```
[the remark this answers is not in the document]
Comment (mrlittle113) [reply api/src/main/java/io/kafbat/ui/exception/GlobalErrorWebExceptionHandler.java:148]: Agree, look much better!

File: api/src/main/java/io/kafbat/ui/exception/GlobalErrorWebExceptionHandler.java [hunk 3]
```diff
@@ -35,6 +36,9 @@
 @Order(Ordered.HIGHEST_PRECEDENCE)
 public class GlobalErrorWebExceptionHandler extends AbstractErrorWebExceptionHandler {
 
+  @Value("${web.exception.include.stacktrace}")
```
[the remark this answers is not in the document]
Comment (mrlittle113) [reply api/src/main/java/io/kafbat/ui/exception/GlobalErrorWebExceptionHandler.java:39]: Can you hind me where to put the config on. 

I saw that **DynamicConfigOperations** load config from **DYNAMIC_CONFIG_PATH_ENV_PROPERTY** which will be overridden on each configuration submission through UI wizard.

And from README I saw that can enable the dynamic through **DYNAMIC_CONFIG_ENABLED: 'true'** on docker compose.

[the remark this answers is not in the document]
Comment (mrlittle113) [reply api/src/main/java/io/kafbat/ui/exception/GlobalErrorWebExceptionHandler.java:39]: Hello @Haarolean, could you help me with this.

> Can you hind me where to put the config on.
> 
> I saw that **DynamicConfigOperations** load config from **DYNAMIC_CONFIG_PATH_ENV_PROPERTY** which will be overridden on each configuration submission through UI wizard.
> 
> And from README I saw that can enable the dynamic through **DYNAMIC_CONFIG_ENABLED: 'true'** on docker compose.

[the remark this answers is not in the document]
Comment (Haarolean) [reply api/src/main/java/io/kafbat/ui/exception/GlobalErrorWebExceptionHandler.java:39]: You can put it temporarily wherever you wish for testing purposes, I like to put mine into `/tmp`.

>which will be overridden on each configuration submission through UI wizard
You have to pass it separately as an arg, like mentioned [here](https://ui.docs.kafbat.io/configuration/configuration-file): `docker run -it -p 8080:8080 -e spring.config.additional-location=/tmp/config.yml -v /tmp/kui/config.yml:/tmp/config.yml ghcr.io/kafbat/kafka-ui` (or an equivalent for a jar file).

[the remark this answers is not in the document]
Comment (mrlittle113) [reply api/src/main/java/io/kafbat/ui/exception/GlobalErrorWebExceptionHandler.java:39]: Thank you, I will try it out

**Comment by devendrapratapsingh** [comment]
@mrlittle113 @Haarolean Is this issue fixed and release in some version? As we are waiting to use this product in our production environment but because of security vulnerability, we can't deploy it. Please provide current status to have it some release?

**Comment by Haarolean** [comment]
> @mrlittle113 @Haarolean Is this issue fixed and release in some version? As we are waiting to use this product in our production environment but because of security vulnerability, we can't deploy it. Please provide current status to have it some release? 

You can see it's closed without being merged due to inactivity. PRs are welcome

**Comment by devendrapratapsingh** [comment]
That I noticed. Do we have any plan or way to merge it in the main? Let me know if I can do something for this, we would like to use this in our production, therefore, I would like to help to merge this into main branch.

**Comment by Haarolean** [comment]
> That I noticed. Do we have any plan or way to merge it in the main? Let me know if I can do something for this, we would like to use this in our production, therefore, I would like to help to merge this into main branch. @mrlittle113 @Haarolean could you please share the plan about this issue?

this is a PR by a third party contributor, I can't attest to their commitment to this issue

**Pull request closed without merging**